projects/step-core/src/lib/modules/resource-input/components/resouce-input/resouce-input.component.ts

```diff
@@ -294,20 +295,32 @@ export class ResourceInputComponent implements OnInit, OnChanges, OnDestroy {
             } else {
               // Creating a new resource
               this.setResourceIdToFieldValue(resourceId);
-              this.lastUploadedResourceId = resourceId;
+              this.uploadedResourceIds.push(resourceId);
             }
           });
       }
     });
   }
 
-  private deleteLastUploadedResource(): void {
-    if (!this.lastUploadedResourceId) {
+  /*
+   * Clean up all resources but the initial one
+   */
+  private deleteUploadedResource(): void {
+    if (!this.uploadedResourceIds || this.uploadedResourceIds.length === 0) {
       return;
     }
 
-    this._augmentedResourcesService.deleteResource(this.lastUploadedResourceId).subscribe(() => {
-      delete this.lastUploadedResourceId;
+    const requests: Observable<any>[] = [];
+    const cleanupIds = Array.from(this.uploadedResourceIds);
+
+    for (const id of cleanupIds) {
+      requests.push(this._augmentedResourcesService.deleteResource(id));
+    }
```

Comment:
Not critical, but personally I would solve it in this way
```suggestion
    const cleanupIds = [...this.uploadedResourceIds];
    const requests = cleanupIds.map((id) => this._augmentedResourcesService.deleteResource(id));
```